shotover/src/hot_reload_client.rs

```diff
@@ -0,0 +1,218 @@
+//This module gives clinet-side implementation for socket handoff as part of hot reloading
+//Client will connect to existing shotovers and requests for FDs
+use crate::hot_reload::{Request, Response};
+use anyhow::{Context, Result};
+use serde::de::DeserializeOwned;
+use std::time::Duration;
+use tokio::io::{AsyncReadExt, AsyncWriteExt, BufReader};
+use tokio::net::UnixStream;
+use tokio::time::timeout;
+use tracing::{debug, info};
+
+pub struct UnixSocketClient {
+    socket_path: String,
+    timeout_duration: Duration,
+}
+
+impl UnixSocketClient {
+    pub fn new(socket_path: String) -> Self {
+        Self {
+            socket_path,
+            timeout_duration: Duration::from_secs(30),
+        }
+    }
+
+    pub fn with_timeout(mut self, timeout: Duration) -> Self {
```

Comment:
Looks like we dont actually call this in shotover, its really used for tests to reduce the timeout time.
Can we mark this as `#[cfg(test)]` to make it obvious that its only used in tests?